### PR TITLE
Improve slice docs

### LINKED ISSUE #29337: API Docs: slices
Part of https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/primitive.slice.html

Lots of things `Deref` to slices, so they're super important.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes  #29337.

r? @steveklabnik

### CODE REVIEW

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -200,28 +200,71 @@ impl<T> [T] {
         core_slice::SliceExt::first(self)
     }
 
-    /// Returns a mutable pointer to the first element of a slice, or `None` if it is empty
+    /// Returns a mutable pointer to the first element of a slice, or `None` if it is empty.
+    ///
+    /// # Example
```
Comment (steveklabnik) [inline src/libcollections/slice.rs]: Can you change these to "Examples" with the s, please?

File: src/libcollections/slice.rs [hunk 2]
```diff
@@ -200,28 +200,71 @@ impl<T> [T] {
...
         core_slice::SliceExt::first_mut(self)
     }
 
     /// Returns the first and all the rest of the elements of a slice.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let x = &[0, 1, 2];
+    /// if let Some((first, elements)) = x.split_first() {
```
Comment (steveklabnik) [inline src/libcollections/slice.rs]: can you add a newline above the `if`, like you did in the first example? Same with tons of them below.

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**